src/engine/Operation.h

```diff
@@ -34,20 +34,50 @@ class Operation {
   // Use existing results if they are already available, otherwise
   // trigger computation.
   shared_ptr<const ResultTable> getResult() const {
-    LOG(DEBUG) << "Try to atomically emplace a new empty ResultTable" << endl;
+    LOG(DEBUG) << "Check cache for Operation result" << endl;
     LOG(DEBUG) << "Using key: \n" << asString() << endl;
     auto [newResult, existingResult] =
         _executionContext->getQueryTreeCache().tryEmplace(asString());
     if (newResult) {
-      LOG(DEBUG) << "We were the first to emplace, need to compute result"
-                 << endl;
+      LOG(DEBUG) << "Not in the cache, need to compute result" << endl;
       // Passing the raw pointer here is ok as the result shared_ptr remains
       // in scope
-      computeResult(newResult.get());
+      try {
+        computeResult(newResult.get());
+      } catch (const ad_semsearch::AbortException& e) {
+        // AbortExceptions have already been printed simply rethrow to
+        // unwind the callstack until the whole query is aborted
```

Comment:
Don't we have to recursively set the `newResult->abort()` also in this recursive call?

Reply:
Yes we absolutely do need to `abort()` here as well, missed that with the last change.